tests/test_shadow_fork.py

```
@@ -76,34 +76,45 @@ def test_create_fails_if_base_branch_missing(self, tmp_path):
+        # However, the original test expected create() to raise RuntimeError.
+        # But create() falls back to other branches if _resolve_base_ref returns them.
+        # If we want to test failure, we need checkout to fail.
+
         with patch.object(fork, "_run_git") as mock_git:
-            mock_git.side_effect = [
-                (True, "main", ""),  # Get current branch
-                (True, "", ""),  # Check for uncommitted changes
-                (False, "", "fatal"),  # Base branch doesn't exist
-            ]
+            # We mock _resolve_base_ref to avoid the loop consuming side_effects
+            with patch.object(fork, "_resolve_base_ref", return_value="nonexistent") as mock_resolve:
```

Ruff (0.14.14)

[error] Local variable `mock_resolve` is assigned to but never used

Remove assignment to unused variable `mock_resolve`

(F841)